gas/config/tc-riscv.c

```diff
@@ -1377,9 +1377,9 @@ riscv_ip (char *str, struct riscv_cl_insn *ip, expressionS *imm_expr,
 		  if (imm_expr->X_op != O_constant
 		      || imm_expr->X_add_number <= 0
 		      || imm_expr->X_add_number >= RISCV_BIGIMM_REACH
-		      || (imm_expr->X_add_number >= RISCV_RVC_IMM_REACH / 2
```

Comment:
I believe this is basically the same issue as #59.  While we made a bad decision about how to expose LUI, we are now stuck with it.